### ISSUE TITLE
fail conversion with anndata 0.7.1

### ISSUE THREAD

**Issue by giovp** [theislab/anndata2ri#43]
`anndata = 0.7.1`
`anndata2ri = 1.0.2`

```python
import anndata2ri

anndata2ri.activate()
%load_ext rpy2.ipython

adata = sc.datasets.visium_sge(sample_id='V1_Breast_Cancer_Block_A_Section_1')

%%R -i adata
adata
```
Takes forever and then error
Error:
```pytb
---------------------------------------------------------------------------
NotImplementedError                       Traceback (most recent call last)
<ipython-input-10-9a903efc8b4e> in <module>
----> 1 get_ipython().run_cell_magic('R', '-i adata_subset', 'adata_subset\n')

~/miniconda3/envs/scdc/lib/python3.7/site-packages/IPython/core/interactiveshell.py in run_cell_magic(self, magic_name, line, cell)
   2360             with self.builtin_trap:
   2361                 args = (magic_arg_s, cell)
-> 2362                 result = fn(*args, **kwargs)
   2363             return result
   2364 

<decorator-gen-414> in R(self, line, cell, local_ns)

~/miniconda3/envs/scdc/lib/python3.7/site-packages/IPython/core/magic.py in <lambda>(f, *a, **k)
    185     # but it's overkill for just that one bit of state.
    186     def magic_deco(arg):
--> 187         call = lambda f, *a, **k: f(*a, **k)
    188 
    189         if callable(arg):

~/miniconda3/envs/scdc/lib/python3.7/site-packages/rpy2/ipython/rmagic.py in R(self, line, cell, local_ns)
    735                         raise NameError("name '%s' is not defined" % input)
    736                 with localconverter(converter) as cv:
--> 737                     ro.r.assign(input, val)
    738 
    739         if args.display:

~/miniconda3/envs/scdc/lib/python3.7/site-packages/rpy2/robjects/functions.py in __call__(self, *args, **kwargs)
    196                 kwargs[r_k] = v
    197         return (super(SignatureTranslatedFunction, self)
--> 198                 .__call__(*args, **kwargs))
    199 
    200 

~/miniconda3/envs/scdc/lib/python3.7/site-packages/rpy2/robjects/functions.py in __call__(self, *args, **kwargs)
    115 
    116     def __call__(self, *args, **kwargs):
--> 117         new_args = [conversion.py2rpy(a) for a in args]
    118         new_kwargs = {}
    119         for k, v in kwargs.items():

~/miniconda3/envs/scdc/lib/python3.7/site-packages/rpy2/robjects/functions.py in <listcomp>(.0)
    115 
    116     def __call__(self, *args, **kwargs):
--> 117         new_args = [conversion.py2rpy(a) for a in args]
    118         new_kwargs = {}
    119         for k, v in kwargs.items():

~/miniconda3/envs/scdc/lib/python3.7/functools.py in wrapper(*args, **kw)
    838                             '1 positional argument')
    839 
--> 840         return dispatch(args[0].__class__)(*args, **kw)
    841 
    842     funcname = getattr(func, '__name__', 'singledispatch function')

~/miniconda3/envs/scdc/lib/python3.7/site-packages/rpy2/robjects/numpy2ri.py in nonnumpy2rpy(obj)
    130         # For now, go with the default_converter.
    131         # TODO: the conversion system needs an overhaul badly.
--> 132         return ro.default_converter.py2rpy(obj)
    133     else:
    134         # The conversion module was "activated"

~/miniconda3/envs/scdc/lib/python3.7/functools.py in wrapper(*args, **kw)
    838                             '1 positional argument')
    839 
--> 840         return dispatch(args[0].__class__)(*args, **kw)
    841 
    842     funcname = getattr(func, '__name__', 'singledispatch function')

~/miniconda3/envs/scdc/lib/python3.7/site-packages/rpy2/robjects/conversion.py in _py2rpy(obj)
    172     raise NotImplementedError(
    173         "Conversion 'py2rpy' not defined for objects of type '%s'" %
--> 174         str(type(obj))
    175     )
    176 

NotImplementedError: Conversion 'py2rpy' not defined for objects of type '<class 'NoneType'>'
```

**Comment by giovp** [comment]
This happens with standard scRNA-seq data as well, not specific to spatial data.

**Comment by flying-sheep** [comment]
Can’t reproduce. I get

```
“Error in validObject(.Object) :
  invalid class "dgCMatrix" object: row indices are not sorted within columns

So I fix it with `adata.X.sort_indices()`, and it works.

**Comment by flying-sheep** [comment]
fixed by #52
